jupysql_plugin/widgets/connections.py

```diff
@@ -1,82 +1,217 @@
+
     # this was added in jupysql 0.10.0
     from sql._current import _get_sql_magic
+
+    # this was renamed in jupysql 0.10.0
+    from sql.parse import connection_str_from_dsn_section
 except (ModuleNotFoundError, ImportError) as e:
     raise ModuleNotFoundError(
         "Your jupysql version isn't compatible with this version of jupysql-plugin. "
         "Please update: pip install jupysql --upgrade"
     ) from e
 
 
-def get_path_to_config_file():
-    """
-    Returns config file path
-    """
-    return _get_sql_magic().dsn_filename
+from jupysql_plugin import exceptions
 
 
-def _get_config_file() -> ConfigParser:
+class ConnectorWidgetManager:
     """
-    Returns current config file
+    Used by the ConnectorWidget to manage database connections and
+    configuration file
     """
-    config = ConfigParser()
 
-    config.read(get_path_to_config_file())
-    return config
+    def get_path_to_config_file(self) -> str:
+        """
+        Returns config file path
+        """
+        return _get_sql_magic().dsn_filename
+
+    def is_config_exist(self) -> bool:
+        """Returns True if the config file exists, False otherwise"""
+        return Path(self.get_path_to_config_file()).is_file()
+
+    def _get_config(self) -> ConfigParser:
+        """
+        Returns current config file
+        """
+        config = ConfigParser()
+
+        config.read(self.get_path_to_config_file())
+        return config
+
+    def _get_connection_string_from_section_in_config_file(
+        self, connection_name
+    ) -> str:
+        """
+        Reads the desired section from the config file and returns the connection
+        string
+        """
+
+        class Config:
+            dsn_filename = Path(self.get_path_to_config_file())
+
+        connection_string = connection_str_from_dsn_section(
+            section=connection_name, config=Config()
+        )
+
+        return connection_string
+
+    def section_name_already_exists(self, connection_name) -> bool:
+        config = ConnectorWidgetManager()._get_config()
+        return connection_name in config.sections()
+
+    def get_connections_from_config_file(self) -> list:
+        """
+        Return the list of connections (dictionaries) from the configuration file
+        """
+        connections = []
+        config = self._get_config()
+
+        def _config_section_to_dict(config, section):
+            d = dict(config.items(section))
+            d["name"] = section
+
+            if "drivername" in d:
+                d["driver"] = d.pop("drivername")
+
+            return d
+
+        connections = [
+            _config_section_to_dict(config, section) for section in config.sections()
+        ]
+
+        return connections
+
+    def save_connection_to_config_file_and_connect(
+        self,
+        connection_data,
+        *,
+        connect=True,
+    ):
+        """
+        Connects to the database specified in the connection_data. If connection
```

Comment:
is this the connection that user defines in the notebook? Maybe we can mention that

Reply:
I'm not following this comment. this is a method that creates a connection with the passed connection data, regardless of where it's coming from